version/version.go

```diff
@@ -1,4 +1,119 @@
 package version
 
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
 // Version is the version of the build.
 const Version = "1.14.1-dev"
+
+// crioVersion is a type for parsing and writing a VersionFile for CRI-O
+type crioVersion struct {
+	Major     string `json:"major"`
+	Minor     string `json:"minor"`
+	SubMinor  string `json:"sub_minor"`
+	Dev       bool   `json:"dev,omitempty"`
+	GitCommit string `json:"git_commit,omitempty"`
+}
+
+// WriteVersionFile writes the version information to a given file
+// file is the location of the old version file
+// gitCommit is the current git commit version. It will be added to the file
+// to aid in debugging, but will not be used to compare versions
+func WriteVersionFile(file string, gitCommit string) error {
```

Comment:
Any chance we could make sure of `https://github.com/blang/semver` that we vendor already?

Reply:
upon looking, yes we can do most of this with semver, but we still need to implement the comparison logic (semver doesn't have the option to ignore Patch, Pre, Build... as we need). :+1: